src/components/ha-picture-upload.ts

```diff
@@ -141,16 +175,58 @@ export class HaPictureUpload extends LitElement {
     }
   }
 
+  private _chooseMedia(): void {
+    showMediaBrowserDialog(this, {
+      action: "pick",
+      entityId: "browser",
+      navigateIds: [
+        { media_content_id: undefined, media_content_type: undefined },
+        {
+          media_content_id: "media-source://image_upload",
+          media_content_type: "app",
+        },
+      ],
+      minimumNavigateLevel: 2,
+      mediaPickedCallback: async (pickedMedia: MediaPickedEvent) => {
+        const id = pickedMedia.item.media_content_id;
+        const stringToRemove = "media-source://image_upload/";
+        if (id.startsWith(stringToRemove)) {
+          const mediaId = id.substr(stringToRemove.length);
+          if (this.crop) {
+            const url = generateImageThumbnailUrl(mediaId, undefined, true);
+            const response = await fetch(url);
+            const data = await response.blob();
```

Comment:
This can fail, please add error handling

Reply:
Can you elaborate a bit what you want to see here? 

So HA gives us the list of images, we click on one, and then... it's not available? What should I do with that, a popup dialog that just says "unknown error"? I would have thought just whatever default logging exception to the console would be sufficient, and how this would usually be handled.

Or what is the specific mode of failure you were thinking of?

Reply:
Sounds good.

If it fails to load use `showAlertDialog` with a text: "Load image preview in crop editor failed" (this is just an example). 

I know this is an edge case but http requests can always fail and showing it only in console won't help the user in the situation.